processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -79,13 +78,42 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    throw new UnsupportedOperationException();
+    return factory.makeValueMatcher(
+        dimension, new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+            return value.equals(Strings.nullToEmpty(fn.apply(Strings.emptyToNull(input))));
```

Comment:
This will fail if value is null as well.  Something like

```
input = Strings.emptyToNull(input);
if (input == null) {
  return value == null;
}

String transformed = fn.apply(input);
if (transformed == null) { 
  return value == null;
}

return transformed.equals(value);
```

And make sure to default empty String to null in the constructor.